### PR TITLE
eda: trends, groupby and calculate growth ratio 

### PR DESCRIPTION
**Pull request by fer78**
In the EDA section, I add three analyses:

* Analyzing the trend of life expectancy and GDP for all countries in the dataset through the time series in the dataset. The band plot shows the central trend with minimum and maximum bands per year, providing an overall picture of the dataset's data.

* A scatter plot of life expectancy and GDP will identify each country according to its relationship between these variables.

* The ratio of life expectancy growth to GDP growth will show to some extent which country made better use of its GDP growth to improve life expectancy.

### CODE REVIEW

File: Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb [hunk 1]
```diff
@@ -141,6 +141,392 @@
...
+    "plt.legend()\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The trend line is positive and shows that there is a constant growth in life expectancy over the analyzed period, suggesting that there are no significant changes in the countries analyzed.\n",
+    "* The lower band is wider throughout the period, which shows that one or several countries that have very low life expectancies improved after the year 2006."
```
Comment (thejjserg) [inline Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb]: "The lower band is wider throughout the period," - I think this sentence can be improved. Perhaps, "The range decreases over time which shows that one or several countries have an overall increase in life expectancies. This is especially shown after the year 2006."

File: Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb [hunk 2]
```diff
@@ -141,6 +141,392 @@
...
+    "plt.ylabel('GDP')\n",
+    "plt.legend()\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The GDP trend also shows growth, although constant, there is a slight decrease in 2018 and then it continues to grow.\n",
```
Comment (thejjserg) [inline Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb]: "The GDP trend also shows growth. The growth is generally constant, but there is a slight decrease in GDP from 2008 to 2009."

Separate the sentence into two sentences and fix the typo.

File: Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb [hunk 3]
```diff
@@ -141,6 +141,392 @@
...
+    "* It would be interesting to analyze what caused the GDP decrease in 2018 in this group.\n",
```
Comment (thejjserg) [inline Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb]: Fix the typo for 2018 to 2008 to 2009.

File: Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb [hunk 4]
```diff
@@ -141,6 +141,392 @@
...
+   "source": [
+    "data_grouped.plot(kind='scatter', x='Life', y='GDP', s=32, alpha=.8, color='blue')\n",
+    "\n",
+    "for index, row in data_grouped.iterrows():\n",
+    "    plt.text(row['Life'], row['GDP'], row['Country'], horizontalalignment='left', verticalalignment='bottom')\n",
+    "\n",
+    "plt.gca().spines[['top', 'right']].set_visible(False)\n",
+    "plt.xlabel('Life')  \n",
+    "plt.ylabel('GDP')   \n",
+    "plt.title('Scatter plot of GDP vs Life with Country Names')  # Añade un título si es necesario\n",
```
Comment (thejjserg) [inline Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb:381]: I'm not sure if using the average gdp and average life expectancy is the best method to visualize the data. In addition, your analysis doesn't mention that we are using the average of the Life and GDP variables. I think Year plays an important aspect of the analysis, so a better comparison might be to take one year and compare Life vs GDP and make a scatterplot(s) that way.

File: Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb [hunk 5]
```diff
@@ -141,6 +141,392 @@
...
+       "5                     China         6.136681      813.418913  0.007544"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Calculate the growth percentage using the first year as the base in relation to the last year and the ratio.\n",
+    "def calculate_growth(df):\n",
```
Comment (thejjserg) [inline Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb:502]: Great function!

File: Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb [hunk 6]
```diff
@@ -141,6 +141,392 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Analysis by countries\n",
+    "\n",
+    "Zimbabwe\n",
+    "\n",
+    "It shows the highest growth in life expectancy (approximately 31.96%) with a also significant GDP growth (approximately 143.72%). It has the highest ratio (0.222), suggesting that the improvement in life expectancy has been considerable even in the face of its economic growth. This is consistent with studies that claim that in countries with low development, variations in GDP have a greater impact on life expectancy because it directly affects literacy and nutrition of the population, which are two essential aspects in disease prevention impacting mortality.\n",
```
Comment (thejjserg) [inline Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb:532]: Good analysis and interpretation. Check your sentences for some typos, and it'll be even better.

Comment (fer78) [reply Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb:532]: Thanks, We will improve it.

File: Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb (hunk 4 above)
Comment (fer78) [reply Life-Expectancy-and-GDP-Starter/life_expectancy_gdp.ipynb:381]: In this chart, I am attempting to show the average position of each country in relation to the two variables and it displays different groups of countries that can be created. Analyzing by groups of similar countries could be an option.

**Comment by thejjserg** [comment]
Reviewed the change. Attempting to merge now.

**Pull request merged**